### ISSUE TITLE
[v3] Implement new Concept Exercise: inheritance

### ISSUE THREAD

**Issue by ErikSchierboom** [exercism/fsharp#867]
This issue describes how to implement the `inheritance` concept exercise for the F# track.

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md)

Please also watch the following video:

- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

## Goal

The goal of this exercise is to teach the student the basics of the Concept of Inheritance in F#.

## Learning objectives

- Know what inheritance is.
- Know when to use inheritance.
- Know how to inherit from a class.
- Know that all types inherit from `object`.
- Know what abstract classes are.
- Know what abstract and default methods are.
- Know how to override methods.
- Know how to using discriminated unions to model hierarchy.

## Out of scope

- Object expressions.

## Concepts

The Concepts this exercise unlocks are:

- `inheritance`: know what inheritance is; know when to use inheritance; know how to inherit from a class; know that all types inherit from `object`; know what abstract classes are; know what abstract and default methods are; know how to override methods; know how to using discriminated unions to model hierarchy;

## Prerequisites

This exercise's prerequisites Concepts are:

- `classes`: defining base and derived classes.
- `discriminated-unions`: defining discriminated unions.

Any data types used in this exercise (e.g. `strings`) should also be added as prerequisites.

### Hints

- [Inheritance][inheritance]: introduction to inheritance using classes.
- [Discriminated union hierarchy][discriminated-union-hierarchy]: shows to do simple inheritance using discriminated unions.

### After

- [Inheritance][inheritance]: introduction to inheritance using classes.
- [Discriminated union hierarchy][discriminated-union-hierarchy]: shows to do simple inheritance using discriminated unions.

## Representer

This exercise does not require any specific representation logic to be added to the [representer][representer].

## Analyzer

This exercise does not require any specific logic to be added to the [analyzer][analyzer].

## Implementing

To implement this exercise, please [follow these instructions](https://github.com/exercism/v3/blob/master/languages/fsharp/reference/implementing-a-concept-exercise.md).

## Help

If you have any questions while implementing the exercise, please post the questions as comments in this issue.

[how-to-implement-a-concept-exercise]: https://github.com/exercism/v3/blob/master/docs/maintainers/generic-how-to-implement-a-concept-exercise.md
[implemented-exercises]: https://github.com/exercism/v3/tree/master/languages/fsharp/exercises/concept/README.md#implemented-exercises
[reference]: https://github.com/exercism/v3/blob/master/reference/concepts/inheritance.md
[analyzer]: https://github.com/exercism/fsharp-analyzer
[representer]: https://github.com/exercism/fsharp-representer
[meta-design]: https://github.com/exercism/v3/blob/master/languages/fsharp/exercises/concept/numbers/.meta/design.md
[meta-config.json]: https://github.com/exercism/v3/blob/master/languages/fsharp/exercises/concept/numbers/.meta/config.json
[docs-concept-exercises]: https://github.com/exercism/v3/blob/master/docs/concept-exercises.md
[inheritance]: https://docs.microsoft.com/en-us/dotnet/fsharp/language-reference/inheritance
[discriminated-union-hierarchy]: https://docs.microsoft.com/en-us/dotnet/fsharp/language-reference/discriminated-unions#using-discriminated-unions-instead-of-object-hierarchies
